### PR TITLE
Add 5 missing attributes to NSURLRequest

### PR DESCRIPTION
**Pull request by danieleggert**
cachePolicy, timeoutInterval, networkServiceType, HTTPBody, and HTTPBodyStream
Rework how attributes are stored.
Refactor header field code into existingHeaderField().

### CODE REVIEW

**Comment by parkera** [comment]
@salgernon - please take a look

**Comment by parkera** [comment]
I think we intentionally left these out, but hopefully @salgernon can comment.

**Comment by danieleggert** [comment]
I think it makes sense to have `cachePolicy`, `timeoutInterval`, and `networkServiceType` on all platforms.

**Comment by parkera** [comment]
If I recall correctly, the reason we left them out is that they are already properties on NSURLSession:

```swift
    /* default cache policy for requests */
    public var requestCachePolicy: NSURLRequestCachePolicy
    
    /* default timeout for requests.  This will cause a timeout if no data is transmitted for the given timeout value, and is reset whenever data is transmitted. */
    public var timeoutIntervalForRequest: NSTimeInterval
    
    /* default timeout for requests.  This will cause a timeout if a resource is not able to be retrieved within a given timeout. */
    public var timeoutIntervalForResource: NSTimeInterval
    
    /* type of service for requests. */
    public var networkServiceType: NSURLRequestNetworkServiceType
    
    /* allow request to route over cellular. */
    public var allowsCellularAccess: Bool
```

**Comment by danieleggert** [comment]
That's why `NSURLSession` has API that takes `NSURL` and API that takes `NSURLRequest`: The API that takes a request let's you override the values set by the `NSURLSessionConfiguration`. As the documentation states:
> By creating a task based on a request object, you can tune various aspects of the task’s behavior, including the cache policy and timeout interval.

File: Foundation/NSURLRequest.swift [hunk 1]
```diff
@@ -364,11 +360,25 @@ public class NSMutableURLRequest : NSURLRequest {
...
+    }
+    public override var timeoutInterval: NSTimeInterval {
+        get { return super.timeoutInterval }
+        set { super.timeoutInterval = newValue }
+    }
+    public override var networkServiceType: NSURLRequestNetworkServiceType {
+        get { return super.networkServiceType }
+        set { super.networkServiceType = newValue }
+    }
+    public override var allowsCellularAccess: Bool {
```
Comment (phausler) [inline Foundation/NSURLRequest.swift]: can this ever be satisfied? It seems like it would require very specific hardware access

Comment (danieleggert) [inline Foundation/NSURLRequest.swift]: I'm fine with leaving `allowsCellularAccess` out.

**Comment by parkera** [comment]
Yes, it was at the request of the owner of this particular API that we do not have this override capability, because they were concerned about the complexity it introduces.

However, since we're not really getting any response on this right now, we may as well add this for parity since it's confusing people right now. Let's leave the cellular one out.

**Comment by danieleggert** [comment]
I just noticed that `HTTPBody` and `HTTPBodyStream` are also missing. I’ll need these for https://github.com/apple/swift-corelibs-foundation/pull/299 in order to implement `POST` and `PUT` requests. Thoughts?

**Comment by danieleggert** [comment]
I've removed `allowsCellularAccess`, but added `HTTPBody` and `HTTPBodyStream`.

**Comment by danieleggert** [comment]
Only after adding `HTTPBody` and `HTTPBodyStream` did I realise that these should not have been added because the `NSURLSession` API has separate methods for uploads (e.g. `POST` / `PUT`). Will remove these again. Sorry for the noise.

**Comment by phausler** [comment]
@swift-ci Please test

**Comment by danieleggert** [comment]
I'll close this.

**Comment by danieleggert** [comment]
C.f. #306

**Pull request closed without merging**